DigitalLearningSolutions.Data/Services/LearningHubSsoSecurityService.cs

```diff
@@ -56,6 +60,19 @@ public bool VerifyHash(string state, string hash)
             return false;
         }
 
+        private byte[] GetSalt(UTF8Encoding encoder)
+        {
+            var salt = encoder.GetBytes(config.GetLearningHubSsoSecretKey());
+
+            if (salt.Length < 8)
+            {
+                logger.LogError("Secret key invalid. The secret key must have a length of at least 8 bytes.");

```

Comment:
We can be more user-friendly - we know we're using UTF8 to convert the string to bytes, so each character is one byte. We can say the key has to be at least 8 characters.